air/src/air/mod.rs

```diff
@@ -283,20 +296,50 @@ pub trait Air: Send + Sync {
     #[allow(unused_variables)]
     fn get_aux_assertions<E: FieldElement<BaseField = Self::BaseField>>(
         &self,
-        aux_rand_elements: &AuxTraceRandElements<E>,
+        aux_rand_elements: &[E],
     ) -> Vec<Assertion<E>> {
         Vec::new()
     }
 
+    // AUXILIARY PROOF VERIFIER
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the [`GkrVerifier`] to be used to verify the GKR proof.
+    ///
+    /// Leave unimplemented if the `Air` doesn't use a GKR proof.
+    fn get_auxiliary_proof_verifier<E: FieldElement<BaseField = Self::BaseField>>(
+        &self,
+    ) -> Self::GkrVerifier {
+        unimplemented!("`get_auxiliary_proof_verifier()` must be implemented when the proof contains a GKR proof");
+    }
+
     // PROVIDED METHODS
     // --------------------------------------------------------------------------------------------
 
+    /// Returns a vector of field elements required for construction of the auxiliary trace segment.
+    ///
+    /// The elements are drawn uniformly at random from the provided public coin.
+    ///
+    /// TODOP: Document how these exclude the Lagrange ones, and how they're put in `AuxRandElements::rand_elements`.
+    fn get_aux_rand_elements<E, R>(&self, public_coin: &mut R) -> Result<Vec<E>, RandomCoinError>
+    where
+        E: FieldElement<BaseField = Self::BaseField>,
+        R: RandomCoin<BaseField = Self::BaseField>,
```

Comment:
Do we not want to pass `Option<LagrangeRandElement>` into this and return `AuxRandElements` from here?

Reply:
I preferred to relieve the user of this boilerplate; we know the user will never need to use the `Option<LagrangeRandElement>`, other than pass it in `AuxRandElements` at the end. I found it cleaner for us to be doing that, and let the user just worry about the "user-defined" aux rand elements